app/javascript/src/ChangePassword/components/ChangePasswordHooks.jsx

```diff
@@ -1,87 +1,103 @@
+// @flow
+
 import { useState } from 'react'
-import { validateForm, validateSingleField } from 'LoginPage/utils/formValidation'
+import validate from 'validate.js' // TODO: including this since validateSingleField and validateForm from LoginPage/utils/formValidation screams for a refactor
+
+type PasswordField = 'user[password]'
+type PasswordConfirmationField = 'user[password_confirmation]'
+type FieldName = PasswordField | PasswordConfirmationField
+type ValidatorName = 'presence' | 'length' | 'equality'
+type ValidatorOption = {message: string, minimun?: number, attribute?: string}
+type Constraints = {[FieldName]: {[ValidatorName]: ValidatorOption}}
+type FieldState = {[FieldName]: string}
+type ValidationErrors = {[FieldName]: string[]}
+type FieldErrorsState = ValidationErrors
+
+const PASSWORD: PasswordField = 'user[password]'
+const PASSWORD_CONFIRMATION: PasswordConfirmationField = 'user[password_confirmation]'
 
-const PASSWORD = 'user[password]'
-const PASSWORD_CONFIRMATION = 'user[password_confirmation]'
-const validationConstraints = {
+// The following validation objects are tied to FormGroups helperTexts, very closed module, not open to change. TODO: REFACTOR those components
+const validationConstraints: Constraints = {
   [PASSWORD]: {
-    presence: true,
-    length: { minimum: 1 }
+    presence: {
+      message: '^isMandatory'
+    },
+    length: {
+      minimum: 1,
+      message: '^password'
+    }
   },
   [PASSWORD_CONFIRMATION]: {
-    presence: true,
-    length: { minimum: 1 },
-    equality: PASSWORD
+    presence: {
+      message: '^isMandatory'
+    },
+    length: {
+      minimum: 1,
+      message: '^isMandatory'
+    },
+    equality: {
+      attribute: PASSWORD,
+      message: '^mustMatch'
+    }
   }
 }
 
-const isFormValid = (formNode) => validateForm(formNode, validationConstraints)
-
-const getPasswordConfirmationError = (errors) => {
-  const passwordConfirmationError = (errors && errors[PASSWORD_CONFIRMATION]) && errors[PASSWORD_CONFIRMATION][0]
-  const passwordsMatchError = passwordConfirmationError && passwordConfirmationError.includes('is not equal to')
-  return {
-    passwordConfirmationError: !!passwordConfirmationError,
-    errorMessage: passwordsMatchError ? 'mustMatch' : 'isMandatory'
-  }
+const fieldsTemplate: FieldState = {
+  [PASSWORD]: '',
+  [PASSWORD_CONFIRMATION]: ''
 }
 
-const comparePasswordsLength = (event) => {
-  const passwordInput = event.currentTarget['user_password']
-  const passwordConfirmationInput = event.currentTarget['user_password_confirmation']
-  return passwordConfirmationInput.value.length >= passwordInput.value.length
+const fieldErrorsTemplate: FieldErrorsState = {
+  [PASSWORD]: [''],
```

Comment:
I think the default state is no errors:
```suggestion
  [PASSWORD]: [],
```